Generate.py

```diff
@@ -256,18 +333,217 @@ def get_choice_legacy(option, root, value=None) -> Any:
+                tokens.append(BinOpToken(val_str))
+                state = "start"
+                i += 1
+            else:
+                raise ValueError(f"unexpected symbol {s[i]} at {i} in: {s}")
+    if paren_count != 0:
+        raise ValueError(f"unclosed parentheses in: {s}")
+    if state != "end":
+        raise ValueError(f"invalid expression: {s}")
+    return tokens
+
+
+def infix_to_postfix(infix: Sequence[AllTokens]) -> Sequence[PostfixTokens]:
+    infix = [OpenParenToken("(")] + list(infix) + [CloseParenToken(")")]
+    op_stack: List[Union[OpenParenToken, BinOpToken]] = []
+    postfix: List[PostfixTokens] = []
+    for token in infix:
+        if isinstance(token, OpenParenToken):
+            op_stack.append(token)
+        elif isinstance(token, CloseParenToken):
+            while len(op_stack):
+                op = op_stack.pop()
+                if isinstance(op, OpenParenToken):
+                    break
+                else:
+                    assert isinstance(assert_type(op, BinOpToken), BinOpToken), f"{op=}"
+                    postfix.append(op)
+        elif isinstance(token, BinOpToken):
+            while True:
+                assert len(op_stack), f"{infix=}"
+                top = op_stack[-1]
+                if _precedence[top.val] >= _precedence[token.val]:
+                    top = op_stack.pop()
+                    assert isinstance(top, BinOpToken), f"{top=} {infix=}"
+                    postfix.append(top)
+                else:
+                    break
+            op_stack.append(token)
+        else:
+            assert isinstance(token, NumberToken) or isinstance(assert_type(token, NameToken), NameToken), f"{token=}"
+            postfix.append(token)
+    return postfix
+
+
+def eval_postfix(postfix: Sequence[PostfixTokens], root: dict, record: ChoiceRecord) -> float:
+    operand_stack: List[float] = []
+    for token in postfix:
+        if isinstance(token, NumberToken):
+            operand_stack.append(token.val)
+        elif isinstance(token, NameToken):
+            if token.val in root:
+                if token.val in record.check_values():
+                    operand_stack.append(record.check_values()[token.val])
+                else:
+                    operand_stack.append(get_choice(token.val, root, record=record))
+                    record.update_value(token.val, operand_stack[-1])
+            else:
+                raise KeyError(f"{token.val} in {postfix} has not been assigned a value in yaml")
+        else:
+            assert isinstance(assert_type(token, BinOpToken), BinOpToken), f"{token=}"
+            if len(operand_stack) < 2:
+                raise ValueError(f"invalid {postfix=}")
+            b = operand_stack.pop()
+            a = operand_stack.pop()
+            if not -1000000 < a < 1000000 or not -1000000 < b < 1000000:
+                raise ValueError(f"Performing math on value outside of valid range: [-1_000_000, 1_000_000]")
```

Comment:
I don't think putting this arbitrary limit on the math has enough value.

A value like 70 could big enough to cause a problem and crash something,
A value like a trillion could be necessary and safe for some games.

The calculation itself is not going to cause any problems no matter how big the numbers are.
I think it's better for the world code that uses the end result to be responsible for the safety.
(I suspect this will be used most often with `Range` type options, and that does bounds checking already.)